README.md

```diff
@@ -107,6 +107,33 @@ DaskTaskRunner(
 )
 ```
 
+### Distributing work across workers
+
+If your task contains a Dask collection, such as a `dask.DataFrame` or `dask.Bag`, to distribute the work across workers and achieve parallel computations, use the `get_dask_client` context manager within your task.
+
+Be mindful of the futures upon `submit` and `compute`. To resolve these futures, call `result`.
+
+```python
+import dask
+from prefect import flow, task
+from prefect_dask import DaskTaskRunner, get_dask_client
+
+@task
+def compute_task():
+    with get_dask_client(timeout="120s") as client:
```

Comment:
How would I use this from within a flow? Might be worth having this detect flow/task run contexts and return the correct client accordingly.

Reply:
Not sure how to incorporate it inside a flow.

I have something like this:
```
@contextmanager
async def get_dask_client(timeout: Optional[Union[int, float, str, timedelta]] = None):
    task_run_context = TaskRunContext.get()
    flow_run_context = FlowRunContext.get()

    client_kwargs = {"timeout": timeout}
    if task_run_context:
        with worker_client(timeout=timeout, separate_thread=False) as client:
            yield client
    elif flow_run_context:
        task_runner = flow_run_context.task_runner
        async with task_runner.start():
            client = task_runner._client
            yield client
```

However, I'm not sure how to make this work for sync and async; if I use `@sync_compatible` I get `Decorated function must be async`

Reply:
The task runner should be started already to be in a flow run context  `async with task_runner.start()` isn't needed. Do we want to return the task runner's async client if the user is in a sync flow?

> However, I'm not sure how to make this work for sync and async; if I use @sync_compatible I get Decorated function must be async even with asynccontextmanager

This is a bug in `sync_compatible`. It should unwrap decorated functions when it checks if they're async. The decorator is sync, which is tricky.

Reply:
I think it's only started when one calls `submit()`

Reply:
Okay I was mistaken; it is started, but I encounter
```
Future: finalize status: cancelled, type: NoneType, key: finalize-8ccf8fd681a0878ac284f866ef97a633
```

```
import time

import dask
from prefect import flow, task
from prefect.context import FlowRunContext
from prefect_dask import DaskTaskRunner, get_dask_client


@flow(task_runner=DaskTaskRunner())
def dask_flow():
    with get_dask_client(timeout=2) as client:
        df = dask.datasets.timeseries("2000", "2005", partition_freq="4w")
        summary_df = client.compute(df.describe())
    return summary_df

dask_flow()
```

with:
```

    task_run_context = TaskRunContext.get()
    flow_run_context = FlowRunContext.get()

    client_kwargs = {"timeout": timeout}
    if task_run_context:
        with worker_client(timeout=timeout, separate_thread=False) as client:
            yield client
    elif flow_run_context:
        task_runner = flow_run_context.task_runner
        yield task_runner._client

```